### ISSUE TITLE
More flexible API or frontend

### ISSUE THREAD

**Issue by Noratrieb** [saethlin/miri-tools#8]
It would be useful to be able to sort the crates (for example by download count).

This could either be implemented here, or another solution that would work as well would be to offer a raw data API, using which you could quickly hack together such sorting or filtering yourself on the command line.

A possible API might look like this:
* `/raw` to get the (shortened, just name/version/error/download count) data of all crates
* `/raw/ub` to only get those with ub
* `/raw/logs` to get the logs of a crate run (just as the raw bytes) (though I'm not sure about the usefulness of this one)

I might be able to hack some if this together tomorrow

**Comment by saethlin** [comment]
You're saying API but this is just a static site served out of an S3 bucket behind CloudFront.

I'm definitely not opposed to having more pages. Just making sure you know how simple the architecture is.

**Comment by Nilstrieb** [comment]
Yes, I am aware of that. Although I don't know whether CloudFront is able to serve static json, that would be useful.

**Comment by Nilstrieb** [comment]
Actually, I don't think I have a need for this anymore right now. Might still be useful for someone in the future though.

**Comment by oli-obk** [comment]
I could hack up a Rust -> wasm thing that gives us a fully interactive website that processes a big json blob which has the actual data (essentially rendering on the fly instead of statically)

**Comment by saethlin** [comment]
:shrug: I'm much more interested in #7 than this, because the primary site how has every crate on crates.io and it's still decently usable. But I'll probably take any PR that looks like I can hack on it later.

**Comment by oli-obk** [comment]
makes sense. Won't spend time on the frontend then.
